lib/guardian/plug.ex

```diff
@@ -201,6 +202,51 @@ if Code.ensure_loaded?(Plug) do
       end
     end
 
+    @spec remember_me(Plug.Conn.t(), module, any, Guardian.Token.claims(), Guardian.opts()) ::
+            Plug.Conn.t()
+    def remember_me(conn, mod, resource, claims \\ %{}, opts \\ []) do
+      with type <- Keyword.get(opts, :token_type, "refresh"),
```

Comment:
I'm not sure this is a good use-case for `with`.

This would be better written (IMO) as:

```ex
def remember_me(conn, mod, resource, claims \\ %{}, opts \\ []) do
  opts = Keyword.put_new(opts, :token_type, "refresh")
  key = key(conn, opts)
    
  case Guardian.encode_and_sign(mod, resource, claims, opts) do
    {:ok, token, new_claims} ->
      put_resp_cookie(conn, key, token, cookie_options(mod, new_claims))
    {:error, _} = err ->
      handle_unauthenticated(conn, err, opts)
  end 
end

defp key(conn, opts) do
  conn
  |> Pipeline.fetch_key(opts)
  |> token_key()
  |> Atom.to_string()
end
```